content/v2.1/helpers/form_helper.md

```diff
@@ -0,0 +1,964 @@
+<input type="text" name="delivery[addresses][][street]" id="delivery-address-1-street" value="">
+```
+
+**Yielding index and the value**
+
+You can get access to the index number and the collection item value, by yielding additional parameters. This way you get control over rendering or tweaking them on the flow.
+
+```ruby
+<% f.fields_for_collection("bill.addresses") do |fa, i, address| %>
+  <div class="form-group">
+    Address id: <%= address.id %>
+    <%= fa.label("street") %>
+    <%= fa.text_field("street", data: {index: i.to_s}) %>
+  </div>
+<% end %>
+```
+
+```html
+<div class="form-group">
+  Address id: 23
+  <label for="bill-addresses-0-street">Street</label>
+  <input type="text" name="bill[addresses][][street]" id="bill-addresses-0-street" value="5th Ave" data-index="0">
+</div>
+<div class="form-group">
+  Address id: 42
+  <label for="bill-addresses-1-street">Street</label>
+  <input type="text" name="bill[addresses][][street]" id="bill-addresses-1-street" value="4th Ave" data-index="1">
+</div>
+```
+
+### label
+Returns a label tag for the given field name, with a humanized version of the field name as the tag's content.
+
+**Usage**
+```ruby
+<%= f.label("book.extended_title") %>
+# => <label for="book-extended-title">Extended title</label>
+```
+
+**Adding HTML attibutes**
+
+```ruby
+<%= f.label("book.title", class: "form-label") %>
+# => <label for="book-title" class="form-label">Title</label>
+```
+
+**Custom content value**
+
+You may specify the `for:` attribute, and pass the label's content as a first argument, to gain the full control over the rendered HTML.
+
+```ruby
+<%= f.label("Title", for: "book.extended_title") %>
+# => <label for="book-extended-title">Title</label>
+
+f.label("book.extended_title", for: "ext-title")
+# => <label for="ext-title">Extended title</label>
+```
+
+**Giving value in a block**
+
+You may also provide the content's value in the block, which allows you to include other HTML tags if needed.
+```ruby
+<%= f.label for: "book.free_shipping" do %>
+  Free shipping
+  <abbr title="optional" aria-label="optional">*</abbr>
+<% end %>
+
+# =>
+<label for="book-free-shipping">
+  Free shipping
+  <abbr title="optional" aria-label="optional">*</abbr>
+</label>
+```
+### fieldset
+Returns a fieldset tag. It is useful to group related items together.
+
+```ruby
+<%= f.fieldset do %>
+  <%= f.legend("Author") %>
+  <%= f.label("author.name") %>
+  <%= f.text_field("author.name") %>
+<% end %>
+
+```
+
+```html
+<fieldset>
+  <legend>Author</legend>
+  <label for="book-author-name">Name</label>
+  <input type="text" name="book[author][name]" id="book-author-name" value="">
+</fieldset>
+```
+
+Keep in mind that this is only a visual improvement and does not affect the data structure being sent to the server.
+### check_box
+Returns the tags for a checkbox.
+
+When editing a resource, the form automatically assigns the `checked` HTML attribute for the checkbox tag.
+
+**Usage**
+
+```ruby
```

Comment:
@swilgosz All the examples below aren't consistent with the article so far. You had an Erb snippet with a corresponding HTML one.

Reply:
Yeah, I need to admit that I got a bit lazy at some point and for me it was way easier to keep HTML with the same snippet.

As I've left the inconsistency, which approach do you think is better?